pennylane_lightning/src/tests/Test_Cache.cpp

```diff
@@ -0,0 +1,135 @@
+#include <catch2/catch.hpp>
+
+#include "Cache.hpp"
+
+using namespace Pennylane::Util;
+
+namespace {
+
+TEST_CASE("LRU_Cache", "[Caching_Indices]") {
+    LRU_cache<std::pair<const std::vector<size_t>, size_t>, std::vector<size_t>>
+        cache_container{10};
+
+    REQUIRE(cache_container.size() == 0);
+    REQUIRE(cache_container.capacity() == 10);
+
+    std::vector<std::pair<const std::vector<size_t>, size_t>> keys = {
+        {{1, 1, 1, 1}, 1},  {{1, 1, 1, 2}, 2},  {{1, 1, 2, 1}, 3},
+        {{1, 2, 1, 1}, 4},  {{1, 2, 1, 3}, 5},  {{1, 2, 3, 1}, 6},
+        {{1, 2, 3, 4}, 7},  {{1, 1, 1, 1}, 8},  {{1, 1, 1, 2}, 9},
+        {{1, 1, 2, 1}, 10}, {{1, 2, 1, 1}, 11}, {{1, 2, 1, 3}, 12},
+        {{1, 2, 3, 1}, 13}, {{1, 2, 3, 4}, 14}};
+
+    size_t number_of_keys = keys.size();
+
+    std::vector<std::vector<size_t>> indices = {
+        {1, 1, 1, 1},     {3, 3, 3, 3},     {5, 5, 5, 5},     {7, 7, 7, 7},
+        {9, 9, 9, 9},     {11, 11, 11, 11}, {13, 13, 13, 13}, {15, 15, 15, 15},
+        {17, 17, 17, 17}, {19, 19, 19, 19}, {21, 21, 21, 21}, {23, 23, 23, 23},
+        {25, 25, 25, 25}, {27, 27, 27, 27}};
+
+    size_t nn = 0;

```

Comment:
What does `nn` refer to here? Is there a more descriptive name for this variable we could use? Maybe it is obvious, but I am not sure yet.